core/src/main/scala/kafka/log/FileMessageSet.scala

```diff
@@ -117,7 +117,13 @@ class FileMessageSet private[kafka](@volatile var file: File,
     new FileMessageSet(file,
                        channel,
                        start = this.start + position,
-                       end = math.min(this.start + position + size, sizeInBytes()))
+                       end = {
+                         // Handle the integer overflow
+                         if (this.start + position + size < 0)
```

Comment:
Theoretically it is possible. But if it is a log segment it seems not possible because the ```this.start``` would be 0, so it is essentially ```position + size```. I don't know if we ever call ```read()``` on a sliced FileMessageSet.